splash/tests/test_execute.py

```diff
@@ -1809,3 +1811,172 @@ def test_get_perf_stats(self):
+                              'outer': '1366x768',
+                              'client': '2000x2000'})
+
+    def test_viewport_overwrites_client_size(self):
+        self.assertSizeAfter('splash:set_window_size("1000x1000");'
+                             'splash:set_viewport("2000x2000")',
+                             {'inner': '2000x2000',
+                              'outer': '1000x1000',
+                              'client': '2000x2000'})
+
+    def test_window_size_overwrites_client_size(self):
+        self.assertSizeAfter('splash:set_viewport("2000x2000");'
+                             'splash:set_window_size("1000x1000");',
+                             {'inner': '1000x1000',
+                              'outer': '1000x1000',
+                              'client': '1000x1000'})
+
+    def test_window_size_validation(self):
+        cases = [
+            ('', 'Invalid window size format: '),
+            ('foobar', 'Invalid window size format: foobar'),
+            ('full', 'Invalid window size format: full'),
+            ('123', 'Invalid window size format: 123'),
+            ('123x', 'Invalid window size format: 123x'),
+            ('x', 'Invalid window size format: x'),
+            ('x123', 'Invalid window size format: x123'),
+            ('0x123', 'Window size is out of range'),
+            ('-1x123', 'Window size is out of range'),
+            ('99999x123', 'Window size is out of range'),
+            ('123x0', 'Window size is out of range'),
+            ('123x-1', 'Window size is out of range'),
+            ('123x99999', 'Window size is out of range'), ]
+
+        def run_test(size_str):
+            self.get_dims_after('splash:set_window_size(%s)' % repr(size_str))
+
+        for size_str, errmsg in cases:
+            self.assertRaisesRegexp(RuntimeError, errmsg,
+                                    run_test, size_str)
+
+    def test_viewport_validation(self):
+        cases = [
+            ('', 'Invalid viewport format: '),
+            ('foobar', 'Invalid viewport format: foobar'),
+            ('123', 'Invalid viewport format: 123'),
+            ('123x', 'Invalid viewport format: 123x'),
+            ('x', 'Invalid viewport format: x'),
+            ('x123', 'Invalid viewport format: x123'),
+            ('0x123', 'Viewport is out of range'),
+            ('-1x123', 'Viewport is out of range'),
+            ('99999x123', 'Viewport is out of range'),
+            ('123x0', 'Viewport is out of range'),
+            ('123x-1', 'Viewport is out of range'),
+            ('123x99999', 'Viewport is out of range'), ]
+
+        def run_test(size_str):
+            self.get_dims_after('splash:set_viewport(%s)' % repr(size_str))
+
+        for size_str, errmsg in cases:
+            self.assertRaisesRegexp(RuntimeError, errmsg,
+                                    run_test, size_str)
+
+    def test_viewport_full(self):
+        w = int('1366x768'.split('x')[0])
```

Comment:
isn't it a constant? And if you want to check for the default value explicitly then it is better to use defaults.WINDOW_SIZE